### ISSUE TITLE
Error: "Too many open files" when having ~1070 entries in "done" folder

### ISSUE THREAD

**Issue by pjotrek-b** [ArkThis/cinbox#20]
The following error message shows up, with around 326 items and 744 logfiles in the `done` folder:

```
[2024-01-26T10:57:42] N  Removing finished Items older than 7 days...
[2024-01-26T10:57:46] ERROR  Error during Inbox execution:
[2024-01-26T10:57:46] ERROR  Failed to open directory "/mnt/inbox/auto/audio/done"
[2024-01-26T10:57:46] ERROR  ArkThis\Helper::getFolderListing(/mnt/inbox/auto/audio/done): failed to open dir: Too many open files[/opt/cinbox-2.0.2-RC4/src/lib/Helper.php:374]
```

What's odd is, that none of these files should be "open" - as the "Helper::getFolderListing()" method should only be listing entries.

**Comment by pjotrek-b** [comment]
I was able to reproduce this error on my development system:
```
Listing item folders in './todo'...
PHP Warning:  ArkThis\Helper::getFolderListing(./todo): failed to open dir: Too many open files in arkthis.github/cinbox/src/lib/Helper.php on line 371
```

**Comment by pjotrek-b** [comment]
Unfortunately, I relied on being able to simply use the SplFileInfo objects as a copy of the filesystem metadata (and not keeping any handle open), so I'm accessing SplFileInfo methods/properties of the generated folder listings, which makes it harder to "fix".

A blunt workaround would be to increase the `the maximum number of open file descriptors`.
To see the current limit of max. open files, use: `$ ulimit -n`

On my setup (Debian 10) that returned 1024.

**Comment by pjotrek-b** [comment]
Fixed in 06e5bbd5eacd7e413067e79c1f32ce15a96de4a9:
Switched from keeping an array of SplFileInfo Objects to copying the values in an associative array.
Now, the file-handles are not kept open anymore.

Tested with 2048 items right now.

**Comment by pjotrek-b** [comment]
NOTE:
The directory listing task may still be affected by a similar issue, because it's using a [RecursiveDirectoryIterator](https://www.php.net/manual/en/class.recursivedirectoryiterator.php) - and keeps the SplFileInfo results in an array.
